website/upgrade-guides/0-11.html.markdown

```diff
@@ -93,9 +93,10 @@ situations:
   the descendent and the descendent no longer inherits attributes from
   its ancestor. Instead, each configuration is entirely distinct.
 
-* Only unaliased provider configurations can be automatically inherited from
-  ancestor modules. Aliased providers must be passed explicitly using
-  [the new `providers` attribute](/docs/modules/usage.html#providers-within-modules).
+* If a `provider` block is present in a child module, a provider must be
```

Comment:
It's also possible to put a full configuration in this block, though I suppose we don't really need to cover that here because it's not something we're recommending people do anyway. For the upgrade guide, it's fine to focus on the recommended path and leave the other possibilities for the full docs we link to here.

Reply:
Yeah, I kind of also want that to be mentioned somehow. Maybe start that with `If a provider block with no configuration is present in a child module ,`?

Reply:
I'm nervous about introducing that here because a block "with no configuration" is a new idea in 0.11 which we haven't explained fully at this point in the text. How about:

> If a `provider` block is present in a child module, it must either contain a complete configuration for its associated provider or a configuration must be passed from the parent module using [the new `providers` argument](/docs/modules/usage.html#providers-within-modules). In the latter case, an empty `provider` block is a placeholder that declares that the child module requires a configuration to be passed from its parent.

This way we get to quickly introduce the idea of a placeholder configuration block without dwelling on it too much, and retain the focus on the idea that a configuration must be explicitly passed from the parent, which I think is the main message here.

Reply:
OK, I think we've come to a good solution here. I updated this paragraph with your proposal.